lib/src/model/game/analysis_ctrl.dart

```diff
@@ -0,0 +1,253 @@
+          ),
+    );
+    return AnalysisCtrlState(
+      gameId: id,
+      initialFen: _root.fen,
+      initialPath: UciPath.empty,
+      currentPath: currentPath,
+      root: _root.view,
+      currentNode: current.view,
+      pov: orientation,
+      numCevalLines: kDefaultLines,
+      numCores: maxCores,
+      isEngineEnabled: true,
+      showBestMoveArrow: true,
+      evaluationContext: evalContext,
+    );
+  }
+
+  void toggleLocalEvaluation() {
+    state = state.copyWith(isEngineEnabled: !state.isEngineEnabled);
+    if (state.isEngineEnabled) {
+      _startEngineEval();
+    } else {
+      ref
+          .read(engineEvaluationProvider(state.evaluationContext).notifier)
+          .stop();
+    }
+  }
+
+  void toggleBestMoveArrow() {
+    state = state.copyWith(showBestMoveArrow: !state.showBestMoveArrow);
+  }
+
+  void setCevalLines(int lines) {
+    if (lines > 3) return;
+    ref
+        .read(engineEvaluationProvider(state.evaluationContext).notifier)
+        .multiPv = lines;
+    _startEngineEval();
+    state = state.copyWith(numCevalLines: lines);
+  }
+
+  void setCores(int num) {
+    if (num > maxCores) return;
+    ref.read(engineEvaluationProvider(state.evaluationContext).notifier).cores =
+        num;
+    _startEngineEval();
+    state = state.copyWith(
+      numCores: num,
+    );
+  }
+
+  void onUserMove(Move move) {
+    if (!state.position.isLegal(move)) return;
+    final (newPath, newNode) = _root.addMoveAt(state.currentPath, move);
+    if (newPath != null) {
+      _setPath(newPath, moveAdded: true, newNode: newNode);
+    }
+  }
+
+  void userNext() {
+    if (state.currentNode.children.isEmpty) return;
+    _setPath(
+      state.currentPath + state.currentNode.children.first.id,
+      replaying: true,
+    );
+  }
+
+  void toggleBoard() {
+    state = state.copyWith(pov: state.pov.opposite);
+  }
+
+  void userPrevious() {
+    _setPath(state.currentPath.penultimate, replaying: true);
+  }
+
+  void userJump(UciPath path) {
+    _setPath(path);
+  }
+
+  void _setPath(
+    UciPath path, {
+    Node? newNode,
+    bool replaying = false,
+    bool moveAdded = false,
+  }) {
+    final pathChange = state.currentPath != path;
+    final rootOrBranch = newNode ?? _root.nodeAt(path);
+
+    if (rootOrBranch.runtimeType == Branch) {
+      final currentNode = rootOrBranch as Branch;
+      if (!replaying) {
+        final isForward = path.size > state.currentPath.size;
+        if (isForward) {
+          final isCheck = currentNode.sanMove.san.contains('+');
+          if (currentNode.sanMove.san.contains('x')) {
+            ref
+                .read(moveFeedbackServiceProvider)
+                .captureFeedback(check: isCheck);
+          } else {
+            ref.read(moveFeedbackServiceProvider).moveFeedback(check: isCheck);
+          }
+        }
+      } else {
+        // when replaying moves fast we don't want haptic feedback
```

Comment:
If I understand the code, it's when replaying (fast or not) that haptic feedback is disabled, which is what we want. (I recently changed archived game behaviour that way).

So the comment can be removed.